packages/ui/src/services/popup/global-popup-manager.service.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2023-present DreamNum Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Tools } from '@univerjs/core';
+import type { IBoundRectNoAngle } from '@univerjs/engine-render';
+import { createIdentifier } from '@wendellhu/redi';
+import type { Observable } from 'rxjs';
+import { BehaviorSubject } from 'rxjs';
+
+export interface IPopup {
+    anchorRect: IBoundRectNoAngle;
+    anchorRect$: Observable<IBoundRectNoAngle>;
+    componentKey: string;
+    onMaskClick?: (e: MouseEvent) => void;
+    unitId: string;
+    subUnitId: string;
+    direction?: 'vertical' | 'horizontal';
+    offset?: [number, number];
+}
+
+export interface IGlobalPopupManagerService {
+    addPopup(item: IPopup): string;
+    removePopup(id: string): void;
+    removeAll(): void;
+
+    popups$: Observable<[string, IPopup][]>;
+
+    get popups(): [string, IPopup][];
+}
+
+export const IGlobalPopupManagerService = createIdentifier<IGlobalPopupManagerService>('univer.popup.service');
+
+export class GlobalPopupManagerService implements IGlobalPopupManagerService {
```

Comment:
CanvasPopupService

Reply:
This popup can be use be for every element by given rect position and popup content, not only for canvas.